### ISSUE TITLE
Feature Request: support for `StorageV2` kind in `azurerm_storage_account.account_kind`

### ISSUE THREAD

**Issue by jambt** [terraform-providers/terraform-provider-azurerm#806]
Hi,

Azure now has the Storage Account V2. It would be great if one could create a Storage Account V2 by setting the `azurerm_storage_account.account_kind` attribute to "StorageV2".

This would be similar to the specification of `kind` as described in the [template documentation](https://docs.microsoft.com/en-us/azure/templates/microsoft.storage/storageaccounts#property-values).

**Comment by jambt** [comment]
As a workaround, I tried to use a local provisioner

```
provisioner "local-exec" {
  command = "az storage account update -g ${var.resource_group} -n ${azurerm_storage_account.storage.name} --set kind=StorageV2"
}
```

However, afterwards `azurerm_storage_container` and `azurerm_storage_queue` do not behave as before. Maybe more changes are required also to these resources?

**Comment by tombuildsstuff** [comment]
hey @jambt 

Thanks for opening this feature request

Taking a quick look into this - it appears [support for this is available in version `2017-10-01` of the Azure Storage API](https://github.com/Azure/azure-sdk-for-go/blob/master/services/storage/mgmt/2017-10-01/storage/models.go#L116) which is available in v12.4 of the Azure SDK for Go. As such it appears that we should be able to upgrade to this new API version and add support for this in the near future.

Thanks!

**Comment by tombuildsstuff** [comment]
:wave: hey @jambt 

Just to let you know that support for this was merged in #893 and has just been released in v1.2.0 of the AzureRM Provider - full details of what's included are available here: https://github.com/terraform-providers/terraform-provider-azurerm/blob/v1.2.0/CHANGELOG.md#120-march-02-2018

Thanks!
